### PR TITLE
MTA-3102-app insights section added

### PR DESCRIPTION
**Pull request by vsodhi**
### JIRA:
* [MTA-3102](https://issues.redhat.com/browse/MTA-3102)

New tab insights is added to the static report.

### Preview: 

https://vsodhi.github.io/previews/MTA/MTA-3102.html#_insights
[images are not displaying in the preview for some reason - working on the fix]

Nandini (QE) has approved the changes.

### CODE REVIEW

File: docs/topics/mta-review-reports.adoc [hunk 1]
```diff
@@ -152,6 +155,56 @@ image::3-5-expanded-issue.png[Expanded rule in the issues report]
...
+
+=== Insights
+
+Insights provide helpful information about the technologies used in the application and their usage in the code.
+For example, a rule searching for a deprecated API usage in the code that does not impact the current migration but can be tracked and fixed when needed in the future.
+
+Unlike issues, insights do not need to be fixed for a successful migration. They are generated by any tagging or non-tagging rule and do not have an effort and category assigned. They might have a message and tag.
+
+[NOTE]
+=====
```
Comment (rromannissen) [inline docs/topics/mta-review-reports.adoc:167]: Wording is a bit odd to say that it's just available in the CLI for the moment.

File: docs/topics/mta-review-reports.adoc [hunk 2]
```diff
@@ -152,6 +155,56 @@ image::3-5-expanded-issue.png[Expanded rule in the issues report]
 
 Issues are sorted into four categories by default. Information on these categories is available at ask Category.
 
+
+=== Insights
```
Comment (rromannissen) [inline docs/topics/mta-review-reports.adoc:159]: I'm missing the reference to this feature being available in Tech Preview.

**Review by rromannissen — CHANGES_REQUESTED** [review]

File: docs/topics/mta-review-reports.adoc (hunk 1 above)
Comment (dymurray) [reply docs/topics/mta-review-reports.adoc:167]: I would reword to be "At this time, application insights are viewable in CLI generated analysis reports at this time" or something like that...

File: docs/topics/mta-review-reports.adoc [hunk 3]
```diff
@@ -152,6 +155,56 @@ image::3-5-expanded-issue.png[Expanded rule in the issues report]
...
+For example, a rule searching for a deprecated API usage in the code that does not impact the current migration but can be tracked and fixed when needed in the future.
+
+Unlike issues, insights do not need to be fixed for a successful migration. They are generated by any tagging or non-tagging rule and do not have an effort and category assigned. They might have a message and tag.
+
+[NOTE]
+=====
+Insights are generated automatically, if applicable or present. However, you must run the application analysis by using the CLI because MTA 7.1 does not support their generation through the UI.
+=====
+
+You can view Insights in the Static report under the Insights tab and in the analysis output under the Insights section.
```
Comment (dymurray) [inline docs/topics/mta-review-reports.adoc:171]: `Static` shouldn't be capitalized right?

File: docs/topics/mta-review-reports.adoc [hunk 4]
```diff
@@ -152,6 +155,62 @@ image::3-5-expanded-issue.png[Expanded rule in the issues report]
...
 
+
+=== Insights
+
+:FeatureName: Application Insights
+include::snippets/technology-preview-admonition.adoc[]
+
+Issues are generated by general rules, whereas string tags are generated by tagging rules. String tags indicate the presence of a technology but do not show the code location. With the introduction of Insights, you can see the technology used in the application along with its usage in the code.
+
+For example, a rule searching for a deprecated API usage in the code that does not impact the current migration but can be tracked and fixed when needed in the future.
```
Comment (anarnold97) [inline docs/topics/mta-review-reports.adoc:166]:
```suggestion
For example, a rule searching for deprecated API usage in the code that does not impact the current migration but can be tracked and fixed when needed in the future.

```

File: docs/topics/mta-review-reports.adoc [hunk 5]
```diff
@@ -152,6 +155,62 @@ image::3-5-expanded-issue.png[Expanded rule in the issues report]
...
+
+Unlike issues, insights do not need to be fixed for a successful migration. They are generated by any tagging or non-tagging rule and do not have an effort and category assigned. They might have a message and tag.
+
+[NOTE]
+=====
+Insights are generated automatically if applicable or present. Currently, {ProductShortName} supports generating Insights when application anaylsis is done using CLI.
+=====
+
+You can view Insights under the Insights tab in the Static report.
```
Comment (anarnold97) [inline docs/topics/mta-review-reports.adoc:175]:
```suggestion
You can view Insights under the *Insights* tab in the Static report.

```

File: docs/topics/mta-review-reports.adoc (hunk 2 above)
Comment (vsodhi) [reply docs/topics/mta-review-reports.adoc:159]: incorporated.

File: docs/topics/mta-review-reports.adoc (hunk 1 above)
Comment (vsodhi) [reply docs/topics/mta-review-reports.adoc:171]: reworded.

File: docs/topics/mta-review-reports.adoc (hunk 3 above)
Comment (vsodhi) [reply docs/topics/mta-review-reports.adoc]: done.

File: docs/topics/mta-review-reports.adoc (hunk 4 above)
Comment (vsodhi) [reply docs/topics/mta-review-reports.adoc]: done.

File: docs/topics/mta-review-reports.adoc (hunk 5 above)
Comment (vsodhi) [reply docs/topics/mta-review-reports.adoc]: done.

**Comment by vsodhi** [comment]
All technical and peer review comments incorporated. Need final ack.

**Comment by anarnold97** [comment]
@nachandr - please could you review for Vibhuti?
Thanks

**Review by dymurray — APPROVED** [review]

File: docs/topics/mta-review-reports.adoc [hunk 6]
```diff
@@ -152,6 +155,62 @@ image::3-5-expanded-issue.png[Expanded rule in the issues report]
...
 Issues are sorted into four categories by default. Information on these categories is available at ask Category.
 
+
+=== Insights
+
+:FeatureName: Insights
+include::snippets/technology-preview-admonition.adoc[]
+
+Issues are generated by general rules, whereas string tags are generated by tagging rules. String tags indicate the presence of a technology but do not show the code location. With the introduction of Insights, you can see the technology used in the application along with its usage in the code.
+
```
Comment (nachandr) [inline docs/topics/mta-review-reports.adoc:165]: Include this statement:
Previously a violation generated by a rule with zero effort was listed as an issue in the static report. This is now listed as an insight instead.

Comment (vsodhi) [reply docs/topics/mta-review-reports.adoc:165]: @nachandr fixed. Please ack.

Comment (nachandr) [reply docs/topics/mta-review-reports.adoc:165]: LGTM

**Review by nachandr — APPROVED** [review]

File: docs/topics/mta-review-reports.adoc [hunk 7]
```diff
@@ -152,6 +155,62 @@ image::3-5-expanded-issue.png[Expanded rule in the issues report]
 
 Issues are sorted into four categories by default. Information on these categories is available at ask Category.
 
+
+=== Insights
+
+:FeatureName: Insights
+include::snippets/technology-preview-admonition.adoc[]
+
+Previously a violation generated by a rule with zero effort was listed as an issue in the static report. This is now listed as an insight instead. Issues are generated by general rules, whereas string tags are generated by tagging rules. String tags indicate the presence of a technology but do not show the code location. With the introduction of Insights, you can see the technology used in the application along with its usage in the code.
```
Comment (anarnold97) [inline docs/topics/mta-review-reports.adoc:164]:
```suggestion
Previously, a violation generated by a rule with zero effort was listed as an issue in the static report. This is now listed as an insight instead. Issues are generated by general rules, whereas string tags are generated by tagging rules. String tags indicate the presence of a technology but do not show the code location. With the introduction of Insights, you can see the technology used in the application along with its usage in the code.
```

So pedantic... i annoy myself

File: docs/topics/mta-review-reports.adoc [hunk 8]
```diff
@@ -152,6 +155,62 @@ image::3-5-expanded-issue.png[Expanded rule in the issues report]
...
+=== Insights
+
+:FeatureName: Insights
+include::snippets/technology-preview-admonition.adoc[]
+
+Previously a violation generated by a rule with zero effort was listed as an issue in the static report. This is now listed as an insight instead. Issues are generated by general rules, whereas string tags are generated by tagging rules. String tags indicate the presence of a technology but do not show the code location. With the introduction of Insights, you can see the technology used in the application along with its usage in the code.
+
+For example, a rule searching for deprecated API usage in the code that does not impact the current migration but can be tracked and fixed when needed in the future.
+
+Unlike issues, insights do not need to be fixed for a successful migration. They are generated by any tagging or non-tagging rule and do not have an effort and category assigned. They might have a message and tag.
```
Comment (pranavgaikwad) [inline docs/topics/mta-review-reports.adoc:168]: can we make this simpler and say "generated by any rule that doesn't have a positive effort value"

**Review by pranavgaikwad — COMMENTED** [review]
Is it worth calling out that insights aren't available in the Hub, just in the CLI? Please ignore if this is already a CLI only doc.

File: docs/topics/mta-review-reports.adoc (hunk 8 above)
Comment (vsodhi) [reply docs/topics/mta-review-reports.adoc]: fixed.

**Comment by vsodhi** [comment]
@pranavgaikwad To avoid confusion, it would be appropriate to clearly distinguish where and how users can find insights.

File: docs/topics/mta-review-reports.adoc (hunk 7 above)
Comment (vsodhi) [reply docs/topics/mta-review-reports.adoc]: fixed.
